src/python/grpcio/grpc/_channel.py

```diff
@@ -95,7 +95,11 @@ class _RPCState(object):
     def __init__(self, due, initial_metadata, trailing_metadata, code, details):
         self.condition = threading.Condition()
         # The cygrpc.OperationType objects representing events due from the RPC's
-        # completion queue.
+        # completion queue. If an operation is in `due`, it is guaranteed that a
+        # `operate()` has been called on a corresponding operation. But the
+        # converse is not true. That is -- in the case of failed `operate()`
+        # calls, there may briefly be events in `due` that do not correspond to
+        # operations submitted to Core.
```

Comment:
Maybe we should mention, all read/write to the `self.due` is protected by `self.condition`. There might be transient events, but won't have a race.